docs/DesignDocs/AttestedTLSAPIs.md

```diff
@@ -0,0 +1,758 @@
+      uint8_t** output_attestation_results_buffer,
+      size_t* uint8_t output_attestation_results_buffer_size);
+  ```
+  The API is used to parse the certificate generated by `oe_get_passport_attestation_certificate`
+  or `oe_get_background_check_attestation_certificate`. An alternative design is representing
+  the output buffers in a list of claims (reducing the number of arguments). However, given that the
+  design does not provide APIs to interpret these buffers therefore the decision is not introdcing
+  pre-defined claims types for them.
+
+
+User Experience
+--------
+
+The following examples demonstrate the use of proposed APIs.
+
+- Passport Model
+
+  ```c
+  static const uint8_t inittime_custom_claims[] =
+    { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+      0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f };
+  ...
+
+  // Attester
+  uint8_t* runtime_custom_claims = public_key; /* Enforce the binding */
+  uint8_t* runtime_custom_claims_buffer_size = public_key_size;
+  uint8_t* attestation_results_buffer = NULL;
+  size_t attestation_results_buffer_size = 0;
+
+  // Send the evidence, init-time custom claims, and run-time custom claims
+  // to the verifier. The results are saved in the attestation_results_buffer.
+
+
+  uint8_t* output_certificate = NULL;
+  size_t output_certificate_size = 0;
+
+  ...
+
+  result = oe_get_passport_attestation_certificate(
+       &format,
+       (const unsigned char*)"CN=Open Enclave SDK,O=OESDK TLS,C=US",
+       private_key,
+       private_key_size,
+       public_key,
+       public_key_size,
+       attestation_results_buffer,
+       attestation_results_buffer_size,
+       &output_certificate,
+       &output_certificate_size);
+  OE_CHECK(result);
+
+  ...
+
+  Send output_certificate to the relying party during the TLS handshake
+
+  ...
+  // Relying party (receive input_certificate)
+
+  uint8_t* attestation_results_buffer;
+  size_t attestation_results_buffer_size;
+
+  result = oe_parse_attestation_certificate(
+       input_certificate,
+       input_certificate_size,
+       NULL,
+       NULL,
+       NULL,
+       NULL,
+       NULL,
+       NULL,
+       &attestation_results_buffer,
+       &attestation_results_buffer_size);
+  OE_CHECK(result);
+  
+  ...
+
+  Validate the attestation results
+  Check the public key against the attestation results
+  ...
+
+  ```
+
+- Background-Check Model
+
+  ```c
+  static const uint8_t inittime_custom_claims[] =
+    { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+      0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f };
+  ...
+
+  // Attester
+  uint8_t* runtime_custom_claims = public_key; /* Enforce the binding */
+  uint8_t* runtime_custom_claims_buffer_size = public_key_size;
+  uint8_t* output_certificate = NULL;
+  size_t output_certificate_size = 0;
+
+  ...
+
+  result = oe_get_background_check_attestation_certificate(
+       &format,
+       (const unsigned char*)"CN=Open Enclave SDK,O=OESDK TLS,C=US",
```

Comment:
Does CN have to be this value?

Reply:
No. Just that existing verification APIs require it.